db/seeds.rb

```diff
@@ -1,58 +1,26 @@
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+# Create admin user
 User.create!(email: 'admin@example.com', password: 'password', password_confirmation: 'password', admin: true) if Rails.env.development?
 
-# declaring ingredients and dishes for testing
-ingredients = Ingredient.create([{name: 'flour'}, {name: 'sugar'}, {name: 'butter'}, {name: 'carrots'}, {name: 'spinach'}, {name: 'apple'}, {name: 'chicken'}, {name: 'beef'}, {name: 'eggs'}, {name: 'salt'}, {name: 'pepper'}, {name: 'cinnamon'}])
-dishes = Dish.create([{name: 'spaghetti'}, {name: 'salad'}, {name: 'cereal'}, {name: 'eggs'}, {name: 'lasagna'},{name: 'pizza'}, {name: 'steak'}, {name: 'potatoes'}, {name: 'risotto'}, {name: 'sandwich'}, {name: 'hamburger'}])
-quantity = [' ounces', ' grams','  kilos',' cans']
+# Import the csv table and remove all blank rows
+table = CSV.parse(File.read("db/menus/cycle1.csv"), headers: true, skip_blanks: true) \
+  .delete_if { |row| row.to_hash.values.all?(&:blank?) }
 
+day = 0
+row = 0
 
-#seeding the database with random dishes and ingredient pairs
-seed = Random.new
-x = 1
-while x < 50
+while row < table.length
+  dish_ids = Array.new
 
-	d = dishes[seed.rand(11)].dup
-	i = ingredients[seed.rand(11)].dup
-	i.portion_size = seed.rand(100).to_s + quantity[seed.rand(4)]
-	i.hc_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	i.bb_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	i.buckley_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	d.ingredients.append(i)
-	
-	Menu.add_dishes_to_cycle(x, [d])
-	
-	d = dishes[seed.rand(11)].dup
-	i = ingredients[seed.rand(11)].dup
-	i.portion_size = seed.rand(100).to_s + quantity[seed.rand(4)]
-	d.ingredients.append(i)
-	i = ingredients[seed.rand(11)].dup
-	i.portion_size = seed.rand(100).to_s + quantity[seed.rand(4)]
-	i.hc_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	i.bb_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	i.buckley_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	d.ingredients.append(i)
-	d.ingredients.append(ingredients[seed.rand(11)].dup)
-	Menu.add_dishes_to_cycle(x, [d])
-	
-	d = dishes[seed.rand(11)].dup
-	d.ingredients.append(ingredients[seed.rand(11)].dup)
-	i = ingredients[seed.rand(11)].dup
-	i.portion_size = seed.rand(100).to_s + quantity[seed.rand(4)]
-	i.hc_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	i.bb_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	i.buckley_prep = seed.rand(1000).to_s + quantity[seed.rand(4)]
-	d.ingredients.append(i)
-	i = ingredients[seed.rand(11)].dup
-	i.portion_size = seed.rand(100).to_s + quantity[seed.rand(4)]
-	d.ingredients.append(i)
+  # Each day ends on "CYCLE: xxx"
+  while !table[row][0].include? "CYCLE" 
+    Dish.create({name: table[row][0]})
+    # Add the id for the newly created dish
+    dish_ids.push(Dish.count)
+    row += 1
```

Comment:
Dish.create returns the dish created. Use that dish and add its id, rather than adding a count to dish_ids. In fact, on my db, the dishes table starts from id 2, where as there is an entry in the menu stating that there a dish with id one exists.

Reply:
Try:
``` 
while !table[row][0].include? "CYCLE" 
    d = Dish.create({name: table[row][0]})
    # Add the id for the newly created dish
    dish_ids.push(d.idt)
    row += 1
```

Reply:
Done.